frameworks/kafka/tests/auth.py

```diff
@@ -170,6 +170,10 @@ def write_failed(output) -> bool:
             if "LEADER_NOT_AVAILABLE" in stderr:
                 LOG.warning("Write failed with LEADER_NOT_AVAILABLE")
                 return True
+            if not is_not_authorized(output):
```

Comment:
Note that `output` is a `tuple` and `is_not_authorized` accepts a string. *Yay for interpreted dynamic typed languages*